pkg/server/server.go

```diff
@@ -293,6 +293,54 @@ func mapSourceSpecToReponse(src client.Object) *pb.Source {
 	return source
 }
 
+func (s *applicationServer) AddApplication(ctx context.Context, msg *pb.AddApplicationRequest) (*pb.AddApplicationResponse, error) {
+	appUrl, err := gitproviders.NewNormalizedRepoURL(msg.Url)
+	if err != nil {
+		return nil, grpcStatus.Error(codes.InvalidArgument, fmt.Sprintf("unable to parse app url %q: %s", msg.Url, err))
+	}
+
+	var configUrl gitproviders.NormalizedRepoURL
+	if msg.ConfigUrl != "" {
+		configUrl, err = gitproviders.NewNormalizedRepoURL(msg.ConfigUrl)
+		if err != nil {
+			return nil, grpcStatus.Error(codes.InvalidArgument, fmt.Sprintf("unable to parse config url %q: %s", msg.ConfigUrl, err))
+		}
+	}
+
+	appSrv, err := s.appFactory.GetAppServiceForAdd(ctx, appUrl.String(), configUrl.String(), msg.Namespace, false, false)
```

Comment:
There's a smell here, that you have `false, false`?

Is there a better API call that doesn't need it?

Reply:
> There's a smell here, that you have false, false?

Agreed. Let me see if I can improve it.

Reply:
Changed to use a `struct` as an argument to `GetAppServiceForAdd`. There was a previous effort to refactor the `apputils`, but it ended up causing more problems than it solved and was closed: https://github.com/weaveworks/weave-gitops/pull/788